server/interaction/npc.go

```diff
@@ -0,0 +1,132 @@
+package interaction
+
+import (
+	"github.com/joshprzybyszewski/cribbage/game"
+	"github.com/joshprzybyszewski/cribbage/logic/scorer"
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/utils/rand"
+)
+
+// NPCType is an enum specifying which type of NPCType
+type NPCType int
+
+// Dumb, Simple, and Calculated are supported
+const (
+	Dumb NPCType = iota
+	Simple
+	Calculated
+)
+
+var npcs = [...]npcPlayer{
+	Dumb: {
+		Type: Dumb,
+		id:   `dumbNPC`,
+	},
+	Simple: {
+		Type: Simple,
+		id:   `simpleNPC`,
+	},
+	Calculated: {
+		Type: Calculated,
+		id:   `calculatedNPC`,
+	},
+}
+
+var _ Player = (*npcPlayer)(nil)
+
+type npcPlayer struct {
+	Type NPCType
+
+	id                   model.PlayerID
+	handleActionCallback func(a model.PlayerAction) error
+}
+
+func (npc *npcPlayer) ID() model.PlayerID {
+	return npc.id
+}
+
+func (npc *npcPlayer) NotifyBlocking(b model.Blocker, g model.Game, s string) error {
+	a := npc.buildAction(b, g)
+	return npc.handleActionCallback(a)
+}
+
+// The NPC doesn't care about messages or score updates
+func (npc *npcPlayer) NotifyMessage(g model.Game, s string) error {
+	return nil
+}
+func (npc *npcPlayer) NotifyScoreUpdate(g model.Game, msgs ...string) error {
+	return nil
+}
+
+// NewNPCPlayer creates a new NPC with specified type
+func NewNPCPlayer(n NPCType, cb func(a model.PlayerAction) error) Player {
+	npc := npcs[n]
+	npc.handleActionCallback = cb
+	return &npc
+}
+
+var me game.Player
```

Comment:
Probably shouldn't have a global variable here - how would this perform with multiple games happening at once?